narwhal/primary/src/proposer.rs

```diff
@@ -459,26 +468,29 @@ impl Proposer {
                 debug!("Dag moved to round {}", self.round);
 
                 // Make a new header.
-                match self.make_header().await {
-                    Err(e @ DagError::ShuttingDown) => debug!("{e}"),
-                    Err(e) => panic!("Unexpected error: {e}"),
-                    Ok((header, digests)) => {
-                        let reason = if max_delay_timed_out {
-                            "max_timeout"
-                        } else if enough_digests {
-                            "threshold_size_reached"
-                        } else {
-                            "min_timeout"
-                        };
-
-                        // Save the header
-                        opt_latest_header = Some(header);
-                        header_repeat_timer = Box::pin(sleep(header_resend_timeout));
-
-                        self.metrics
-                            .num_of_batch_digests_in_header
-                            .with_label_values(&[reason])
-                            .observe(digests as f64);
+                let make = rand::random::<usize>() % 100;
```

Comment:
I don't understand the change in the proposer logic (do we have to have it in this PR?): could you please add some comments here?

Reply:
this is also temporary